Spanner/tests/System/BatchTest.php

```diff
@@ -123,81 +133,57 @@ public function testBatch()
         $snapshot->close();
     }
 
-    public function testBatchWithRestrictiveDatabaseRole()
+    public function testBatchWithDbRole()
     {
         // Emulator does not support FGAC
         $this->skipEmulatorTests();
 
-        $error = null;
-        $query = 'SELECT
-                id,
-                decade
-            FROM ' . self::$tableName . '
-            WHERE
-                decade > @earlyBound
-            AND
-                decade < @lateBound';
-
-        $parameters = [
-            'earlyBound' => 1960,
-            'lateBound' => 1980
-        ];
-
-        $resultSet = iterator_to_array(self::$database->execute($query, ['parameters' => $parameters]));
-
-        $batch = self::$client->batch(self::INSTANCE_NAME, self::$dbName, ['databaseRole' => self::$databaseRole]);
-        $string = $batch->snapshot()->serialize();
-
-        $snapshot = $batch->snapshotFromString($string);
-
-        try {
-            $partitions = $snapshot->partitionQuery($query, ['parameters' => $parameters]);
-        } catch (ServiceException $e) {
-            $error = $e;
+        foreach ($this->dbProvider() as $dbProvided) {
```

Comment:
Alternate approach is, add `@dataProvider` annotation on top of test method and the test will automatically see `$dbRole, $expected` arguments.

Reply:
There is a problem with `@dataProvider` and static variables in `setUpBeforeClass`. Basically the `setUpBeforeClass` function gets called after the `@dataProvider` takes values for static variables due to their order in execution. Please ref: [stackoverflow](https://stackoverflow.com/questions/52867554/access-static-variable-from-setupbeforeclass-in-dataprovider-method-in-php-u) for more information.

Reply:
You can call `self::setUpBeforeClass();` in the dataprovider and prevent double setup using a control flag like `$hasSetUp`, like [ValueMapperTest](https://github.com/googleapis/google-cloud-php/blob/4c71429693c5d7e4402e5663ffdfb313c796650b/Firestore/tests/System/ValueMapperTest.php#L31-L37).

Reply:
Done